### DISCUSSION TITLE
Maintaining focus

### DISCUSSION THREAD

**Discussion by GHNewbiee** [lxqt/lxqt#2538] [Q&A]
**Case**: After booting, I open N apps by clicking one-by-one. While apps getting open, I choose to work with a specific one.
**Question**: Can (through an option) this app maintain the focus so that I can continue my work uninterruptedly?
*Tia*

**Comment by tsujan** [comment]
I think you may be talking about "focus stealing prevention". It's a property that some WMs have. You need to check if the WM you use has such a setting and change it.

Please also note that this isn't related to LXQt.

**Reply by GHNewbiee** [reply]
Sorry, "The Lightweight Qt Desktop Environment" does not provide its own WM, or it's not coming with a default one?!

A desktop env does not include a WM?

**Reply by tsujan** [reply]
LXQt doesn't reinvent the wheel. It's modular, meaning that any DE-agnostic WM, compositor,... works with it. That's the case for both X11 and Wayland — actually, we've had a hard time to keep it so for Wayland.

**Reply by GHNewbiee** [reply]
So, it is a matter of X11 or Wayland support and not of a DE. Do I understand correct?

**Reply by tsujan** [reply]
> So, it is a matter of X11 or Wayland support and not of a DE.

I didn't understand this. You asked about window focus, and I replied.

LXQt doesn't and won't have a WM of its own. There are several, good X11 WMs that work will all DEs (Openbox isn't one of them; it's dead).

As for Wayland, the situation is almost like a mess, but recently the light can be seen at the end of the tunnel because of `wlroots`. The goal is making LXQt compatible with all `wlroots`-based Wayland compositors, and it's already done for some components in git or PRs. As far as we've tested various Wayland compositors, LabWC seems to be the best choice.

**Reply by GHNewbiee** [reply]
Ah now it is more clear. X11/Wayland, WMs and DE are different parts. Responsible part for maintaining/stealing focus feature is the WM. So, I need to find a WM which provides the support to the feature I want.

The last paragraph is beyond me to understand. 

*Talfyt*

**Reply by tsujan** [reply]
> X11/Wayland, WMs and DE are different parts. Responsible part for maintaining/stealing focus feature is the WM

Correct.

**Comment by Butterfly** [comment]
Hi.. "Just" a user/lurker here. Try right-clicking your preferred window's icon wherever it (hopefully) is showing up on a "panel" (mine are at the top). Look for something like "Layer" as a right-click option. Expand that out by whatever means are presented then try "Always on  top" if it appears next. I didn't actually activate it, but did find it available for e.g. `xfce4-terminal` and my favored browser. :tada:

**Reply by GHNewbiee** [reply]
The use of "Always on top" is totally different than that I want and describe.
